test/upgrade/2.0.0-2.1.0.js

```diff
@@ -0,0 +1,587 @@
+    exchangeHandler,
+    offerHandler,
+    fundsHandler,
+    disputeHandler,
+    bundleHandler,
+    groupHandler,
+    twinHandler,
+    configHandler;
+  // orchestrationHandler, pauseHandler, metaTransactionsHandler,
+  let mockToken;
+  let snapshot;
+  let protocolDiamondAddress;
+  let mockAuthERC721Contract;
+
+  let DRs = [];
+  let sellers = [];
+  let buyers = [];
+  let agents = [];
+  let offers = [];
+  let exchanges = [];
+  let protocolContractState;
+
+  before(async function () {
+    // Make accounts available
+    [deployer, rando] = await ethers.getSigners();
+
+    // checkout old version
+    const oldVersion = "v2.0.0";
+    console.log(`Checking out version ${oldVersion}`);
+    shell.exec(`git checkout ${oldVersion} contracts`);
+
+    // run deploy suite, which automatically compiles the contracts
+    await hre.run("deploy-suite", { env: "upgrade-test" });
+
+    // Read contract info from file
+    const chainId = (await hre.ethers.provider.getNetwork()).chainId;
+    const contractsFile = readContracts(chainId, "hardhat", "upgrade-test");
+
+    // Get AccessController abstraction
+    const accessControllerInfo = contractsFile.contracts.find((i) => i.name === "AccessController");
+    accessController = await ethers.getContractAt("AccessController", accessControllerInfo.address);
+
+    // Temporarily grant UPGRADER role to deployer account
+    await accessController.grantRole(Role.UPGRADER, deployer.address);
+
+    // Get protocolDiamondAddress
+    protocolDiamondAddress = contractsFile.contracts.find((i) => i.name === "ProtocolDiamond").address;
+
+    // Grant PROTOCOL role to ProtocolDiamond address and renounces admin
+    await accessController.grantRole(Role.PROTOCOL, protocolDiamondAddress);
+
+    // Cast Diamond to interfaces
+    accountHandler = await ethers.getContractAt("IBosonAccountHandler", protocolDiamondAddress);
+    bundleHandler = await ethers.getContractAt("IBosonBundleHandler", protocolDiamondAddress);
+    disputeHandler = await ethers.getContractAt("IBosonDisputeHandler", protocolDiamondAddress);
+    exchangeHandler = await ethers.getContractAt("IBosonExchangeHandler", protocolDiamondAddress);
+    fundsHandler = await ethers.getContractAt("IBosonFundsHandler", protocolDiamondAddress);
+    groupHandler = await ethers.getContractAt("IBosonGroupHandler", protocolDiamondAddress);
+    offerHandler = await ethers.getContractAt("IBosonOfferHandler", protocolDiamondAddress);
+    // orchestrationHandler = await ethers.getContractAt("IBosonOrchestrationHandler", protocolDiamondAddress);
+    twinHandler = await ethers.getContractAt("IBosonTwinHandler", protocolDiamondAddress);
+    // pauseHandler = await ethers.getContractAt("IBosonPauseHandler", protocolDiamondAddress);
+    // metaTransactionsHandler = await ethers.getContractAt("IBosonMetaTransactionsHandler", protocolDiamondAddress);
+    configHandler = await ethers.getContractAt("IBosonConfigHandler", protocolDiamondAddress);
+
```

Comment:
If I uncomment the right away, linter is complaining.
I just prepared them, since I expect them to be used at some point in either one of two skipped contexts:
`📋 New data after the upgrade do not corrupt data from before the upgrade`
`📋 Interactions after the upgrade still work`